### ISSUE TITLE
Preview callback webpanel.handleMessage's message.value.node is always "this"

### ISSUE THREAD

**Issue by jnthngdbt** [tintinweb/vscode-interactive-graphviz#18]
Hi, thanks for the awesome extension. Sorry for the cryptic title.

I am developing an extension that generates a DOT file and previews it using your extension. I want to catch 'onClick' events on nodes to capture the clicked node ID.

I have the following code that correctly hooks to 'onClick' events. 

```
preview(fullname: string, content: string) {
  let callback = (webpanel: any) => this.interactivePreviewWebpanelCallback(webpanel);
  vscode.workspace.openTextDocument(fullname).then( (document) => {
      let args = { document, content, callback };
      vscode.commands.executeCommand("graphviz-interactive-preview.preview.beside", args);
  });
}

interactivePreviewWebpanelCallback(webpanel: any) {
  let handler = (message: any) => { return this.interactivePreviewMessageHandler(message); }
  webpanel.handleMessage = handler;
}

interactivePreviewMessageHandler(message: any) {
  console.log(JSON.stringify(message));

  switch(message.command){
      case 'onClick':
          console.log("onClick");
          break;
      case 'onDblClick':
          console.log("onDblClick");
          break;
      default:
          console.warn('Unexpected command: ' + message.command);
  }
}
```

However, when I try to get the clicked node ID from message.value.node, it is always "this":

`{"command":"onClick","value":{"node":"this"}}`

Do you have any clues?

Thanks,

Jonathan

**Comment by jnthngdbt** [comment]
That would be exactly what I need! Thanks for the quick response.

**Comment by tintinweb** [comment]
released with v0.0.12, cheers 🙌
